src/containerapp/azext_containerapp/_params.py

```diff
@@ -260,6 +260,7 @@ def load_arguments(self, _):
                    validator=validate_build_env_vars, is_preview=True)
         c.argument('custom_location_id', options_list=['--custom-location'], help="Resource ID of custom location. List using 'az customlocation list'.", is_preview=True)
         c.argument('connected_cluster_id', help="Resource ID of connected cluster. List using 'az connectedk8s list'.", configured_default='connected_cluster_id', is_preview=True)
+        c.argument('kind', type=str, help="Set to 'functionapp' to enable built-in support and autoscaling for Azure Functions on Azure Container Apps.", is_preview=True)
```

Comment:
I notice the `kind` already set under 'containerapp create':
```
    with self.argument_context('containerapp create') as c:
        c.argument('kind', type=str, help="Set to 'functionapp' to enable built-in support and autoscaling for Azure Functions on Azure Container Apps.", is_preview=True)

```

Recommend move to `with self.argument_context('containerapp', arg_group='Configuration') as c:`, so they can be used together for command `az containerapp create/up`
```
    with self.argument_context('containerapp', arg_group='Configuration') as c:
        c.argument('kind', type=str, help="Set to 'functionapp' to enable built-in support and autoscaling for Azure Functions on Azure Container Apps.", is_preview=True)
```

Reply:
```suggestion
        c.argument('kind', arg_type=get_enum_type(['functionapp']), help="Set to 'functionapp' to enable built-in support and autoscaling for Azure Functions on Azure Container Apps.", is_preview=True)
```

Reply:
Hi @khushishah513 
Please take a look for this comment from zunli